packages/salesforce-adapter/src/filters/custom_objects.ts

```diff
@@ -354,11 +352,18 @@ const createNestedMetadataInstances = (instance: InstanceElement,
           instance.value[INSTANCE_FULL_NAME_FIELD],
           nestedInstance[INSTANCE_FULL_NAME_FIELD],
         ].join(API_NAME_SEPARATOR)
-        const elemIdName = naclCase(fullName)
+        const instanceName = [
+          naclCase(instance.value[INSTANCE_FULL_NAME_FIELD]),
+          naclCase(nestedInstance[INSTANCE_FULL_NAME_FIELD]),
+        ].join('_')
```

Comment:
not sure if this is a problem, but this will cause the instance name to behave a bit strangely and I think it might have the potential to create a conflict because there will be one `_` that is not escaped (the one used to `join` the parts at the end)
maybe it would be better to change this to
```suggestion
        const instanceName = naclCase([
          instance.value[INSTANCE_FULL_NAME_FIELD],
          nestedInstance[INSTANCE_FULL_NAME_FIELD],
        ].join('_'))
```
or, if we include the repetition with the code above, maybe change to
```typescript
const nameParts = [apiName(instance), nestedInstance[INSTANCE_FULL_NAME_FIELD]]
const fullName = nameParts.join(API_NAME_SEPARATOR)
const saltoName = naclCase(nameParts.join('_'))
```